src/screens/MainApp/DiagnoseRequest/navigators/MainNavigator.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { createAppContainer } from 'react-navigation'
+import { createStackNavigator } from 'react-navigation-stack'
+import MainScreen from '../screens/MainScreen'
+import DescriptionRequest from '../screens/DescriptionRequest'
+import FinishRequest from '../screens/FinishRequest'
+import ImageReview from '../screens/ImageReview'
+import NoPhotoDisclaimer from '../screens/NoPhotoDisclaimer'
+
+const Navigator = createStackNavigator(
+  {
+    MainScreen,
+    DescriptionRequest,
+    FinishRequest,
+    ImageReview,
+    NoPhotoDisclaimer
+  },
+  {
+    initialRouteName: 'MainScreen'
+  }
+)
+
+const MainNavigator = createAppContainer(Navigator)
+
+MainNavigator.navigationOptions = ({ navigation }) => {
+  let tabBarVisible = true
+
+  if (navigation.state.index > 0) {
+    tabBarVisible = false
+  }
+
+  return { tabBarVisible }
+}
```

Comment:
🤔 why is the tab bar hidden if the `navigation.state.index` is `> 0`? This code is not clear enough. Try using a function/variable for the conditions, so that you can use a descriptive name: (the following name is only an example)
```javascript
let tabBarVisible = true
const isFirstScreen = navigation.state.index === 0

if (!isFirstScreen) tabBarVisible = false

return { tabBarVisible }
```
When I read the code above I interpret `The tab bar is visible on all screens, except the first one`. I don't need to know the specifics of the condition, but I can find it quickly if I want.